asserts/assertstest/assertstest.go

```diff
@@ -141,3 +143,185 @@ func GPGImportKey(homedir, armoredKey string) {
 		panic(fmt.Errorf("cannot import test key into GPG setup at %q: %v (%q)", homedir, err, out))
 	}
 }
+
+// A SignerDB can sign assertions using its key pairs.
+type SignerDB interface {
+	Sign(assertType *asserts.AssertionType, headers map[string]string, body []byte, keyID string) (asserts.Assertion, error)
+}
+
+// NewAccount creates an account assertion for username, it fills in values for other missing headers as needed. It panics on error.
+func NewAccount(db SignerDB, username string, headers map[string]string, keyID string) *asserts.Account {
```

Comment:
Is it really worth splitting `username` out as a separate param? Strictly speaking, `username` is optional anyway and I suspect its value is irrelevant in many tests.